src/measure_theory/decomposition/jordan.lean

```diff
@@ -0,0 +1,381 @@
+finite measures. -/
+@[ext] structure jordan_decomposition (α : Type*) [measurable_space α] :=
+(pos_part neg_part : measure α)
+[pos_part_finite : finite_measure pos_part]
+[neg_part_finite : finite_measure neg_part]
+(mutually_singular : pos_part ⊥ₘ neg_part)
+
+attribute [instance] jordan_decomposition.pos_part_finite
+attribute [instance] jordan_decomposition.neg_part_finite
+
+namespace jordan_decomposition
+
+open measure vector_measure
+
+variable (j : jordan_decomposition α)
+
+instance : has_zero (jordan_decomposition α) :=
+{ zero := ⟨0, 0, mutually_singular.zero⟩ }
+
+instance : inhabited (jordan_decomposition α) :=
+{ default := 0 }
+
+instance : has_neg (jordan_decomposition α) :=
+{ neg := λ j, ⟨j.neg_part, j.pos_part, j.mutually_singular.symm⟩ }
+
+/-- The signed measure associated with a Jordan decomposition. -/
+def to_signed_measure : signed_measure α :=
+j.pos_part.to_signed_measure - j.neg_part.to_signed_measure
+
+lemma to_signed_measure_zero : (0 : jordan_decomposition α).to_signed_measure = 0 :=
+begin
+  ext1 i hi,
+  erw [to_signed_measure, to_signed_measure_sub_apply hi, sub_self, zero_apply],
+end
+
+lemma to_signed_measure_neg : (-j).to_signed_measure = - j.to_signed_measure :=
+begin
+  ext1 i hi,
+  rw [neg_apply, to_signed_measure, to_signed_measure,
+      to_signed_measure_sub_apply hi, to_signed_measure_sub_apply hi, neg_sub],
+  refl,
+end
+
+/-- A Jordan decomposition provides a Hahn decomposition. -/
+lemma exists_compl_positive_negative :
+  ∃ S : set α, measurable_set S ∧
+    j.to_signed_measure ≤[S] 0 ∧ 0 ≤[Sᶜ] j.to_signed_measure ∧
+    j.pos_part S = 0 ∧ j.neg_part Sᶜ = 0 :=
+begin
+  obtain ⟨S, hS₁, hS₂, hS₃⟩ := j.mutually_singular,
+  refine ⟨S, hS₁, _, _, hS₂, hS₃⟩,
+  { refine restrict_le_restrict_of_subset_le _ _ (λ A hA hA₁, _),
+    rw [to_signed_measure, to_signed_measure_sub_apply hA,
+        show j.pos_part A = 0, by exact nonpos_iff_eq_zero.1 (hS₂ ▸ measure_mono hA₁),
+        ennreal.zero_to_real, zero_sub, neg_le, zero_apply, neg_zero],
+    exact ennreal.to_real_nonneg },
+  { refine restrict_le_restrict_of_subset_le _ _ (λ A hA hA₁, _),
+    rw [to_signed_measure, to_signed_measure_sub_apply hA,
+        show j.neg_part A = 0, by exact nonpos_iff_eq_zero.1 (hS₃ ▸ measure_mono hA₁),
+        ennreal.zero_to_real, sub_zero],
+    exact ennreal.to_real_nonneg },
+end
+
+end jordan_decomposition
+
+namespace signed_measure
+
+open measure vector_measure jordan_decomposition classical
+
+variables {s : signed_measure α} {μ ν : measure α} [finite_measure μ] [finite_measure ν]
+
+/-- Given a signed measure `s`, `s.to_jordan_decomposition` is the Jordan decomposition `j`,
+such that `s = j.to_signed_measure`. This property is known as the Jordan decomposition
+theorem, and is shown by
+`measure_theory.signed_measure.to_signed_measure_to_jordan_decomposition`. -/
+def to_jordan_decomposition (s : signed_measure α) : jordan_decomposition α :=
```

Comment:
You should be able to show `to_jordan_composition_zero` too, right?

Reply:
I've added the `neg` version as well, though it's at the end of the file since it needs `to_signed_measure_invjective`.

Reply:
That was how I expected you to prove it anyway. It might make sense to prove the zero version the same way.